spec/models/upload_certificate_event_spec.rb

```diff
@@ -124,6 +127,13 @@
         expect(event).to_not be_valid
         expect(event.errors[:certificate]).to eql [t('certificates.errors.small_key')]
       end
+
+      it 'component ends up with 2 certs if an UploadCertificatesEvents is invalid' do
+        valid_event1 = UploadCertificateEvent.create(usage: CERTIFICATE_USAGE::SIGNING, value: good_cert_value, component: msa_component)
+        valid_event2 = UploadCertificateEvent.create(usage: CERTIFICATE_USAGE::SIGNING, value: good_cert_value, component: msa_component)
+        invalid_event = UploadCertificateEvent.create(usage: CERTIFICATE_USAGE::SIGNING, value: "", component: msa_component)
```

Comment:
```suggestion
        invalid_event = UploadCertificateEvent.create(usage: CERTIFICATE_USAGE::SIGNING, value: good_cert_value, component: msa_component)
        expect(valid_event1).to be_valid
        expect(valid_event1).to be_valid
        expect(invalid_event).to_not be_valid
```

Reply:
By giving the invalid_event a good value, the valid events get the errors fail to publish. Any ideas why?

Reply:
hm, that suggests something is not validating properly? giving a bad value is cheating :D as it fails on the value, not the number of certs...

Reply:
you'll have to debug it probably, why it gets past the validation...

Reply:
So the issue was that when it went through the validation it was checking that `component.enabled_signing_certificates.count <= 2`. So on the third event that returns true as it does equal 2 so therefore adds a third one. Just just changing that to `< 2`

Reply:
good we do the testing!